packages/monorepo-builder/packages/split/src/PackageToRepositorySplitter.php

```diff
@@ -80,6 +82,16 @@ public function splitDirectoriesToRepositories(
             $tag = $this->gitManager->getMostRecentTag($rootDirectory);
         }
 
+        // Ensure branch exists on remotes/origin
+        if ($this->gitManager->doesBranchExistOnRemote($branch) === false) {
+            $message = sprintf(
+                'Given branch "%s" does\'t exist on remotes/origin. Please ensure to push it before running split.',
```

Comment:
Avoid escaping ''''. It can be easily forgot or quotes switched

Reply:
Why does have to branch exist on remote?

Reply:
Because if the branch doesn't exist on remote then the split command crashes

Reply:
Oh, that's annoying :/. Kind of kills the *split explicit branch* feature

When I push locally any branch (e.g. here on Github), it is created in the remove repository for me.

What way could work for this case so people don't have to create all the branches on remote?

Reply:
@TomasVotruba Are you sure you're not using any tool for git locally? Because usually when you try to push a branch that doesn't exist on the remote git gives you that

```bash
git checkout -b doesnt-exist
git push

fatal: The current branch doesnt-exist has no upstream branch.
To push the current branch and set the remote as upstream, use

    git push --set-upstream origin doesnt-exist
```

But looking at the example that's what we could do, if the branch doesn't exist on the remote we automatically do a `git push --set-upstream` this way people don't have to push the branch manually. I will update the PR and you tell me

Reply:
I think I do this first then forget:

```bash
git push --set-upstream origin
```

Could we do the same here to easy the frustration?

I imagine if I want to push `beta` branch into 20 packages, I would not enjoy going to all 20 repositories manually and creating `beta` branch there :D Kind of kills the DX/UX value of this tool.

---

> I will update the PR and you tell me

Awesome thank you! If this is too much for one PR, just tell me *no* and we can focus on it in next PR

Reply:
@TomasVotruba I've already pushed the change but obviously I've upset the coding standards because I went to fast..! 😄 

> I imagine if I want to push beta branch into 20 packages, I would not enjoy going to all 20 repositories manually and creating beta branch there :D Kind of kills the DX/UX value of this tool.

No no, so the issue is only with the monorepo, if the branch isn't on the monorepo it breaks but for the multirepos it works perfectly after that. But even if the issue is only on the monorepo I agree that having the automatic push --set-upstream makes the feature even better, should have thought about it from the start 😛